src/cheri_insts.sail

```diff
@@ -108,6 +110,48 @@ function clause execute(CJAL(imm, cd)) = {
   }
 }
 
+val handle_cjalr_isentry_ptp : (regidx, regidx) -> Retired effect { escape, rmem, rmemt, rreg, wmv, wmvt, wreg }
+function handle_cjalr_isentry_ptp(cd, cs) = {
+  let cs_val = C(cs);
+  let idc = unsealCap(cs_val);
+  let idcaddr = cs_val.address;
+
+  // tag and otype has already been checked by caller
+  if not (idc.permit_load) | not (idc.permit_load_cap) then {
+    handle_cheri_reg_exception(CapEx_PermitLoadViolation, cs);
+    RETIRE_FAIL
+  } else if not(inCapBounds(idc, idcaddr, cap_size)) then {
+    handle_cheri_reg_exception(CapEx_LengthViolation, cs);
+    RETIRE_FAIL
+  } else if not(is_aligned_addr(idcaddr, cap_size)) then {
+    handle_mem_exception(idcaddr, E_Load_Addr_Align());
+    RETIRE_FAIL
+  } else {
+    match load_cap_via_cap(0b0 @ cs, cs_val, idcaddr) {
+      None() => RETIRE_FAIL,
+      Some(v) => {
+        if not (v.tag) then {
+          handle_cheri_reg_exception(CapEx_TagViolation, cs);
+          RETIRE_FAIL
+        } else if not (v.permit_execute) then {
+          handle_cheri_reg_exception(CapEx_PermitExecuteViolation, cs);
+          RETIRE_FAIL
+        } else {
```

Comment:
Need to handle v being sealed (especially if it points to a sentry, which is a bit redundant but IMO should be allowed). At which point this basically looks like a copy of CJALR, except it rejects nested indirect sentries, so probably worth sharing code.